src/System.CodeDom/tests/CodeCollections/CodeCollectionTestBase.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.CodeDom.Tests
+{
+    public abstract class CodeCollectionTestBase<TCollection, TItem> where TCollection: class where TItem : class, new()
+    {
+        public abstract TCollection Ctor();
```

Comment:
Nit: Do all of the collections have a parameterless ctor?  If so, we could use a new() constraint on TCollection as well to avoid needing this abstract method.